### PR TITLE
libspdm: Support generating direct manifest

### PR DESCRIPTION
**Pull request by alistair23**
This supports generating a direct manifest from an indirect manifest. This is based on the original discussion at [1].

1: https://gitlab.com/twittner/minicbor/-/merge_requests/39

### CODE REVIEW

File: src/libspdm/manifest.rs [hunk 1]
```diff
@@ -153,3 +161,218 @@ pub fn decode_cbor_manifest(buffer: &[u8], use_pretty: bool) -> Result<Vec<u8>,
...
+        .value_mut()
+        .tagged_evidence
+        .first_mut()
+        .unwrap()
+        .value_mut()
+        .ce_ev_triples;
+    let measurement_maps = &mut ce_ev_triples.ce_membership_triples.1;
+
+    for measurement_map in measurement_maps {
+        println!("measurement_map: {:?}", measurement_map);
```
Comment (twilfredo) [inline src/libspdm/manifest.rs:285]: any reason to not use logging macros `debug!(), info!()` here?

File: src/libspdm/manifest.rs [hunk 2]
```diff
@@ -153,3 +161,218 @@ pub fn decode_cbor_manifest(buffer: &[u8], use_pretty: bool) -> Result<Vec<u8>,
...
+                0x87 => {
+                    // svn
+                    let value = measurement_record[measurement_offset + 0] as i64
+                        | (measurement_record[measurement_offset + 1] as i64) << 8
+                        | (measurement_record[measurement_offset + 2] as i64) << 16
+                        | (measurement_record[measurement_offset + 3] as i64) << 24;
+
+                    measurement_map.mval.svn = Some(Tagged::<552, i64>::new(value));
+                }
+                0x80 | 0xFF => {
```
Comment (twilfredo) [inline src/libspdm/manifest.rs:357]: Thought on using enums for the indexes?

File: src/libspdm/manifest.rs [hunk 3]
```diff
@@ -153,3 +161,218 @@ pub fn decode_cbor_manifest(buffer: &[u8], use_pretty: bool) -> Result<Vec<u8>,
...
+) -> Result<SpdmToc, minicbor::decode::Error> {
+    let mut spdm_toc: SpdmToc<'_> = minicbor::decode(&measurement_manifest)?;
+    let ce_ev_triples = &mut spdm_toc
+        .value_mut()
+        .tagged_evidence
+        .first_mut()
+        .unwrap()
+        .value_mut()
+        .ce_ev_triples;
+    let measurement_maps = &mut ce_ev_triples.ce_membership_triples.1;
```
Comment (twilfredo) [inline src/libspdm/manifest.rs:282]: is this basically a _map_ of all of the measurements?

**Review by twilfredo — APPROVED** [review]
Nice! makes it pretty easy fetch and read all of the measurements.

File: src/libspdm/manifest.rs (hunk 3 above)
Comment (alistair23) [reply src/libspdm/manifest.rs:282]: Yep!

**Comment by alistair23** [comment]
Addressed your comments

**Review by twilfredo — APPROVED** [review]

**Pull request merged**